Unlink any temporary files left around from the Batch Server

Krebrov001 opened the pull request:
This includes shared memory files, fifo files, and the lock file.

Signed-off-by: Konstantin Rebrov <konstantin.rebrov@intel.com>

- Relates to #2272
- Fixes #2272

Krebrov001 committed:
Unlink any temporary files left around from the Batch Server

This includes shared memory files, fifo files, and the lock file.

Signed-off-by: Konstantin Rebrov <konstantin.rebrov@intel.com>

cmcantalupo commented on service/geopmdpy/system_files.py:
```diff
@@ -682,7 +685,21 @@ def remove_batch_server(self, client_pid):
         self._sessions[client_pid].pop('batch_server')
         self._update_session_file(client_pid)
-        # TODO: Unlink any shared memory files left around
+
```
Please add some tests.  This code should not do any pattern matching.  The file paths should be deterministic based on the batch server PID.  As it it written now this could delete files that are unrelated to the particular batch server that ended.

Krebrov001 committed:
Change the algorithm that looks for temp files

Make it look for the files with the batch server PID in the name,
instead of looking for any files matching the temp file format,
which would  pick up files from other batch servers.

Signed-off-by: Konstantin Rebrov <konstantin.rebrov@intel.com>

Krebrov001 committed:
Added the test `TestActiveSessions.remove_batch_server`

Tests that the batch server was removed, and all the temporary files
were sucessfully deleted.

Signed-off-by: Konstantin Rebrov <konstantin.rebrov@intel.com>

cmcantalupo commented on service/geopmdpy/system_files.py:
```diff
@@ -679,10 +682,42 @@ def remove_batch_server(self, client_pid):
+
+        # Unlink any temporary files left around from the Batch Server
+        # This includes shared memory files, fifo files, and the lock file.
+        if (os.path.exists(self._LOCK_PATH)):
```
Cleaning the lock file is done when the session ends (line 687).  That code checks if the session that obtained the lock file is still alive, and it is not it is removed.  The code here will remove the lock file regardless of which session owns it.  This may remove a lock created by a process other than the batch server.  The removal of this file must depend on the PID that has ended.

cmcantalupo commented on service/geopmdpy_test/TestActiveSessions.py:
```diff
@@ -437,6 +437,77 @@ def test_batch_server_bad_service_restart(self):
+        sess_path = f'{self._TEMP_DIR.name}/geopm-service'
+        full_file_path = os.path.join(sess_path, f"session-{client_pid}.json")
+
+        lock_path = os.path.join(sess_path, "CONTROL_LOCK")
```
remove the test around the write lock.

cmcantalupo reviewed (commented):
Looks good.  We don't want to unlink the lock file if it exists any time a batch server ends, so please remove this feature, otherwise that should solve the issue.
